app/models/manageiq/providers/redhat/infra_manager/event_parsing/strategies/v4.rb

```diff
@@ -27,18 +27,23 @@ def self.event_to_hash(event, ems_id = nil)
         :vm_ems_ref          => vm_ref,
         :host_ems_ref        => ems_ref_from_object_in_event(event.host),
         :ems_cluster_ems_ref => ems_ref_from_object_in_event(event.cluster),
-        :vm_location         => vm_location(event.data_center, vm_ref)
       }
+
+      vm_location(hash, event.data_center, vm_ref)
     end
 
-    def self.vm_location(dc, vm_ref)
+    def self.vm_location(hash, dc, vm_ref)
```

Comment:
I think this should either have another name or we should add the "return nil if vm_ref.nil?" without the adding the hash as a parameter and clear the vm_location key from the hash in the calling function in case its value is nil.
But to me this name does not correspond to a method that returns a hash with a lot of other attributes in addition to location, and some times does not return any location at all.

Reply:
I will update the name. Please note that we do not want to have nil value for this key so returning nil is not an option.